contracts/PositionToken.sol

```diff
@@ -64,8 +65,16 @@ contract PositionToken is ERC721, ERC721Burnable, AccessControl {
         return _exists(positionTokenId);
     }
 
-    /* Override required by Solidity: */
-    function supportsInterface(bytes4 interfaceId) public view override(ERC721, AccessControl) returns (bool) {
+    /* Overrides required by Solidity: */
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal virtual override(ERC721, ERC721Enumerable) {
```

Comment:
We need to add non reentrancy here ([link](https://app.gitbook.com/o/B7Qi247Rg5Gidqy0rmTN/s/fcHzCkSayJoCWD74wFz1/infrastructure-reviews/20220420-apy-finance-and-zellic) to gitbook note from talk with auditors, scroll to "Re-entrance/Design
")